qsp_transfer.py

```diff
@@ -0,0 +1,108 @@
+import os
+import sys
+import traceback
+import json
+import boto3
+import base64
+import time
+import re
+import logging
+
+from ftplib import FTP
+from datetime import datetime, timedelta
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+def get_ftp_credentials():
+    client = boto3.client('secretsmanager')
+    get_secret_value_response = client.get_secret_value(SecretId=os.environ['SECRET_NAME'])
+
+    if 'SecretString' in get_secret_value_response:
+        secret = get_secret_value_response['SecretString']
+    else:
+        secret = base64.b64decode(get_secret_value_response['SecretBinary'])
+
+    return json.loads(secret)
+
+def create_data_file(date):
+    if not re.match(r"\d{2}-\d{2}-\d{4}", date):
+        raise Exception(f"Date parameter value '{date}' does not match expected format DD-MM-YYYY")
+
+    log_group = os.environ['LOG_GROUP_NAME']
+    start_time = int(time.mktime(time.strptime(f"{date} 00:00:00", '%d-%m-%Y %H:%M:%S'))) * 1000
+    end_time = int(time.mktime(time.strptime(f"{date} 23:59:59", '%d-%m-%Y %H:%M:%S'))) * 1000
+
+    logging.info(f"Using log group name: {log_group}")
+    logging.info(f"Using epoch start time in milliseconds: {start_time}")
+    logging.info(f"Using epoch end time in milliseconds: {end_time}")
+
+    events = []
+
+    client = boto3.client('logs')
+    response = client.filter_log_events(
+        logGroupName = log_group,
+        startTime = start_time,
+        endTime = end_time
+    )
+
+    events = events + response['events']
```

Comment:
I'd use `append`. I.e. `events.append(response['events'])`

Reply:
Agreed, this is currently allocating a new list every time rather than appending to the existing one. 👍🏻